src/trade-matcher/trade.ts

```diff
@@ -33,8 +33,18 @@ const getEncodedTradeBySourceAmount = (
   y: BigNumber,
   z: BigNumber,
   A: BigNumber,
-  B: BigNumber
+  B: BigNumber,
+  simulated = false
 ): BigNumber => {
+  if (simulated) {
```

Comment:
can both A and z be 0? because it will be div by 0

Reply:
Of course, that's why we can allow ourselves to disable trading via this setup instead of allocating an additional flag in every strategy order.